CMakeLists.txt

```diff
@@ -200,6 +200,7 @@ else ()
     ADD_CHECK_CXX_COMPILER_FLAG(CMAKE_CXX_FLAGS CXX_WARN_RANGE_LOOP_ANALYSIS -Wrange-loop-analysis)
     ADD_CHECK_CXX_COMPILER_FLAG(CMAKE_CXX_FLAGS CXX_WARN_TAUTOLOGICAL_ZERO_COMPARE -Wtautological-unsigned-zero-compare)
     ADD_CHECK_CXX_FLAG(CMAKE_CXX_FLAGS CXX_WNO_CLOBBERED -Wno-clobbered)
+    ADD_CHECK_CXX_FLAG(CMAKE_CXX_FLAGS CXX_WNNO_CAST_FUNCTION_TYPE -Wno-cast-function-type)
```

Comment:
There's a typo: `WNNO`, double `N`.

Shouldn't this also only be enabled for mingw builds?

Reply:
Indeed there is an extra N in there, and yes it probably does only need to be enabled for mingw.